h2integrate/converters/wind/wind_plant.py

```diff
@@ -48,10 +52,136 @@ def setup(self):
+        )
+        heights = np.repeat(bounding_heights, len(self.data_to_field_number))
+        field_number_to_data = {v: k for k, v in self.data_to_field_number.items()}
+        fields = np.tile(list(field_number_to_data.keys()), len(bounding_heights))
+        n_timesteps = int(self.options["plant_config"]["plant"]["simulation"]["n_timesteps"])
+
+        resource_data = np.zeros((n_timesteps, len(fields)))
+        cnt = 0
+        for height, field_num in zip(heights, fields):
+            rounding_precision = data_to_precision[field_number_to_data[field_num]]
+            resource_key = f"{field_number_to_data[field_num]}_{int(height)}m"
+            if resource_key in wind_resource_data:
+                resource_data[:, cnt] = wind_resource_data[resource_key].round(rounding_precision)
+            else:
+                if any(
+                    field_number_to_data[field_num] in c for c in list(wind_resource_data.keys())
+                ):
+                    data_heights = [
+                        float(c.split("_")[-1].replace("m", "").strip())
+                        for c in list(wind_resource_data.keys())
+                        if field_number_to_data[field_num] in c
+                    ]
+                    if len(data_heights) > 1:
+                        nearby_heights = [
+                            self.calculate_bounding_heights_from_resource_data(
+                                hub_ht,
+                                wind_resource_data,
+                                resource_vars=[field_number_to_data[field_num]],
+                            )
+                            for hub_ht in data_heights
+                        ]
+                        nearby_heights = functools.reduce(operator.iadd, nearby_heights, [])
+                        nearby_heights = list(set(nearby_heights))
+                        if len(nearby_heights) > 1:
+                            height_diff = [
+                                np.abs(valid_height - height) for valid_height in nearby_heights
+                            ]
+                            closest_height = nearby_heights[np.argmin(height_diff).flatten()[0]]
+                            resource_key = (
+                                f"{field_number_to_data[field_num]}_{int(closest_height)}m"
+                            )
+
+                        else:
+                            resource_key = (
+                                f"{field_number_to_data[field_num]}_{int(nearby_heights[0])}m"
+                            )
+
+                    else:
+                        resource_key = f"{field_number_to_data[field_num]}_{int(data_heights[0])}m"
+                    if resource_key in wind_resource_data:
+                        resource_data[:, cnt] = wind_resource_data[resource_key].round(
+                            rounding_precision
+                        )
+            cnt += 1
```

Comment:
The logic in this section is tough to follow at a glance due to the nested if statements and multiple list comprehensions. Could you please add explanatory comments throughout or maybe refactor parts of the list logic to be more digestible? Let me know if you want to do this together, I know this feedback is potentially vague.